field_utils/FieldBundleDelta.F90

```diff
@@ -0,0 +1,306 @@
+! This class is to support propagation of time-dependent Field
+! attributes across couplers as well as to provide guidance to the
+! containt Action objects on when to recompute internal items.
+
+#include "MAPL_Exceptions.h"
+module mapl3g_FieldBundleDelta
+   use mapl3g_LU_Bound
+   use mapl3g_FieldDelta
+   use mapl3g_InfoUtilities
+   use mapl_FieldUtilities
+   use mapl_FieldPointerUtilities
+   use mapl3g_esmf_info_keys
+   use mapl_ErrorHandling
+   use mapl_KeywordEnforcer
+   use esmf
+   implicit none (type, external)
+   private
+
+   public :: FieldBundleDelta
+
+   type :: FieldBundleDelta
+      private
+      type(FieldDelta) :: field_delta ! constant across bundle
+      real(ESMF_KIND_R4), allocatable :: interpolation_weights(:)
+   contains
+      procedure :: initialize_bundle_delta
+      generic :: initialize => initialize_bundle_delta
+      procedure :: update_bundle
+      procedure :: reallocate_bundle
+   end type FieldBundleDelta
+
+
+   interface FieldBundleDelta
+      procedure new_FieldBundleDelta
+      procedure new_FieldBundleDelta_field_delta
+   end interface FieldBundleDelta
+
+contains
+
+   function new_FieldBundleDelta(fieldCount, geom, typekind, num_levels, units, interpolation_weights) result(bundle_delta)
+      type(FieldBundleDelta) :: bundle_delta
+      integer, optional, intent(in) :: fieldCount
+      type(ESMF_Geom), optional, intent(in) :: geom
+      type(ESMF_TypeKind_Flag), optional, intent(in) :: typekind
+      integer, optional, intent(in) :: num_levels
+      character(*), optional, intent(in) :: units
+      real(ESMF_KIND_R4), intent(in), optional :: interpolation_weights(:)
+
+      associate (field_delta => FieldDelta(geom=geom, typekind=typekind, num_levels=num_levels, units=units))
+        bundle_delta = FieldBundleDelta(field_delta, fieldCount, interpolation_weights)
+      end associate
+
+   end function new_FieldBundleDelta
+
+   function new_FieldBundleDelta_field_delta(field_delta, fieldCount, interpolation_weights) result(bundle_delta)
+      type(FieldBundleDelta) :: bundle_delta
+      type(FieldDelta), intent(in) :: field_delta
+      integer, optional, intent(in) :: fieldCount
```

Comment:
Does this get used?

Reply:
Probably not yet.    It is just a wrapper for the full constructor (or vice versa).  When deployed in couplers we can circle back to see which constructors here should survive.